server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/GetSnapshotsResponse.java

```diff
@@ -51,24 +52,31 @@
             (p, c) -> p.map(HashMap::new, ElasticsearchException::fromXContent),
             new ParseField("failures")
         );
+        GET_SNAPSHOT_PARSER.declareStringOrNull(ConstructingObjectParser.optionalConstructorArg(), new ParseField("next"));
     }
 
     private final List<SnapshotInfo> snapshots;
 
     private final Map<String, ElasticsearchException> failures;
 
-    public GetSnapshotsResponse(List<SnapshotInfo> snapshots, Map<String, ElasticsearchException> failures) {
+    @Nullable
+    private final String next;
```

Comment:
I think we should update `equals` and `hashCode` to account for the new field too.

Reply:
++